internal/config/from_env.go

```diff
@@ -57,15 +55,28 @@ func (c *envConfig) GetWithSource(hostname, key string) (string, string, error)
 
 func (c *envConfig) CheckWriteable(hostname, key string) error {
 	if hostname != "" && key == "oauth_token" {
-		envName := GITHUB_TOKEN
-		if ghinstance.IsEnterprise(hostname) {
-			envName = GITHUB_ENTERPRISE_TOKEN
+		if _, env, found := AuthTokenFromEnv(hostname); found {
+			return fmt.Errorf("read-only token in %s cannot be modified", env)
 		}
+	}
+
+	return c.Config.CheckWriteable(hostname, key)
+}
 
-		if os.Getenv(envName) != "" {
-			return fmt.Errorf("read-only token in %s cannot be modified", envName)
+func AuthTokenFromEnv(hostname string) (string, string, bool) {
```

Comment:
I'm not sure if there's value in returning a `found` value. The caller can simply check `token != ""` at call site, like before.

With the current implementation, it looks like it's possible for `found` to be positive, but the value of the token to be empty.